tests/unit/test_api_database_comprehensive.py

```
@@ -0,0 +1,215 @@
+
+        with get_connection() as conn:
+            assert conn == mock_conn
+
+        mock_conn.close.assert_not_called()
+
+
+class TestCleanupMemoryConnection:
+    """Test cases for _cleanup_memory_connection function."""
+
+    @patch("api.database._MEMORY_CONNECTION")
+    def test_cleanup_closes_memory_connection(self, mock_conn):
```

Ruff (0.14.11)

Unused method argument: `mock_conn`

(ARG002)